presto-main/src/main/java/com/facebook/presto/testing/LocalQueryRunner.java

```diff
@@ -772,14 +775,23 @@ public PlanNodeId getNextId()
                 stageExecutionDescriptor,
                 subplan.getFragment().getRoot(),
                 subplan.getFragment().getPartitioningScheme().getOutputLayout(),
-                plan.getTypes(),
                 subplan.getFragment().getTableScanSchedulingOrder(),
                 outputFactory,
-                new TaskExchangeClientManager(ignored -> {
-                    throw new UnsupportedOperationException();
-                }),
-                createTableWriteInfo(streamingSubPlan, metadata, session),
-                ImmutableMap.of());
+                new RemoteSourceFactory()
```

Comment:
I am wondering if we should have a `TestingRemoteSourceFactory`?

Reply:
LocalQueryRunner doesn't add remote exchanges, as it is intended to be run only in a local, inmemory mode. Thus the `RemoteSourceFactory` should never be used. I'm not sure if `TestingRemoteSourceFactory` is the best name for a factory that always throws the `UnsupportedOperationException`. I was thinking whether i can give it a meaningful name, but i didn't manage to comeup with anything reasonable. Since it is used only in one place i simply decided to inline it.